pkg/debug/spec.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/sh
+
+if [ "$(uname -m)" = x86_64 ]; then
+   ARCH=2
+else
+   ARCH=4
+fi
+
+if [ -e /dev/xen ]; then
+   CPUS=$(eve exec xen-tools xl info | grep nr_cpus | cut -f2 -d:)
+   MEM=$(( $(eve exec xen-tools xl info | grep total_memory | cut -f2 -d:) / 1024 ))
+else
+   CPUS=$(grep -c '^processor.*' < /proc/cpuinfo)
+   MEM=$(awk '/MemTotal:/ { print int($2 / 1048576); }' < /proc/meminfo)
+fi
+
+DISK=$(lsblk -b  | grep disk | awk '{ total += $4; } END { print int(total/(1024*1024*1024)); }')
+
+cat <<__EOT__
+{
+  "arch": $ARCH,
+  "productURL": "$(cat /persist/status/hardwaremodel || cat /config/hardwaremodel)",
+  "productStatus": "production",
+  "attr": {
+    "memory": "${MEM}G",
+    "storage": "${DISK}G",
+    "Cpus": "${CPUS}"
+  },
+  "logo": {
+    "logo_back":"/workspace/spec/logo_back_.jpg",
+    "logo_front":"/workspace/spec/logo_front_.jpg"
+  },
+  "ioMemberList": [
+__EOT__
+
+#enumerate GPUs
+ID=""
+for VGA in $(lspci -D  | grep VGA | cut -f1 -d\ ); do
+cat <<__EOT__
+    {
+      "ztype": 7,
+      "phylabel": "VGA${ID}",
+      "assigngrp": "VGA${ID}",
+      "phyaddrs": {
+        "PciLong": "${VGA}"
+      },
+      "logicallabel": "VGA${ID}",
+      "usagePolicy": {}
+    },
+__EOT__
+    ID=$(( ${ID:-0} + 1 ))
+done
+
+#enumerate USB
+ID=""
+for USB in $(lspci -D  | grep USB | cut -f1 -d\ ); do
```

Comment:
Maybe docs/comments that this includes the USB controllers but not the indivual ports aka receptacles.